Quadriga/src/main/java/edu/asu/spring/quadriga/service/workbench/mapper/impl/ProjectShallowMapper.java

```diff
@@ -240,4 +240,50 @@ public IProject getProjectDetails(ProjectDTO projectDTO) throws QuadrigaStorageE
 		return projectList;
 	}
 	
+	/**
+	 * {@inheritDoc}
+	*/
+	@Override
+	@Transactional
+	public IProject getProjectDetailsForSearch(ProjectDTO projectDTO, String pattern) throws QuadrigaStorageException{
+		IProject projectProxy = null;
+		if(projectDTO != null){
+			projectProxy = new ProjectProxy(projectManager);
+			projectProxy.setProjectId(projectDTO.getProjectid());
+			projectProxy.setProjectName(projectDTO.getProjectname());
+			projectProxy.setDescription(searchLines(projectDTO.getDescription(), pattern));
+			projectProxy.setProjectAccess(EProjectAccessibility.valueOf(projectDTO.getAccessibility()));
+			projectProxy.setUnixName(projectDTO.getUnixname());
+			projectProxy.setOwner(userDeepMapper.getUser(projectDTO.getProjectowner().getUsername()));
+			projectProxy.setCreatedBy(projectDTO.getCreatedby());
+			projectProxy.setCreatedDate(projectDTO.getCreateddate());
+			projectProxy.setUpdatedBy(projectDTO.getUpdatedby());
+			projectProxy.setUpdatedDate(projectDTO.getUpdateddate());
+		}
+		return projectProxy;
+	}
+
+	/**
+	 * Returns formated String for description and pattern
+	 * @param description project description
+	 * @param pattern search string
+	 * @return formated description
+	 */
+	private String searchLines(String description, String pattern) {
+		String[] temp = description.split("\\.");
+		StringBuffer finalDescription = new StringBuffer();
+		int linesCount = 0;
+		for(int i=0; i<temp.length; i++){
+			if(temp[i].toLowerCase().matches(".*"+pattern.toLowerCase()+".*") && linesCount < 4){
+				if(i+1 < temp.length && temp[i+1].toLowerCase().matches(".*"+pattern.toLowerCase()+".*")){

```

Comment:
Instead of this you can use Java 8's join method:
https://docs.oracle.com/javase/8/docs/api/java/lang/String.html
Will make your code much cleaner.